### PR TITLE
add support for dynamic chart versions

### PR DESCRIPTION
**Pull request by a-hat**

### CODE REVIEW

File: pkg/helmt/helmt_test.go [hunk 1]
```diff
@@ -186,8 +186,8 @@ func TestHelmTemplate(t *testing.T) {
 			},
 			expectedCommands: []string{
 				"helm version",
-				"helm fetch --repo https://kubernetes-charts.storage.googleapis.com --version 2.0.0 jenkins",
-				"helm template something jenkins-2.0.0.tgz --include-crds --skip-tests --output-dir .",
+				"helm fetch --repo https://kubernetes-charts.storage.googleapis.com --version 2.0.0 --destination /temp/helmt-123 jenkins",
+				"helm template something /temp/helmt-123/temp/helmt-123/chart-1.0.0.tgz --include-crds --skip-tests --output-dir .",
```
Comment (torstenwalter) [inline pkg/helmt/helmt_test.go:190]: Is there one temp/helmt-123 to much?

File: pkg/helmt/helmt_test.go [hunk 2]
```diff
@@ -197,8 +197,8 @@ func TestHelmTemplate(t *testing.T) {
 			},
 			expectedCommands: []string{
 				"helm version",
-				"helm fetch --repo https://hub.syncier.cloud/chartrepo/library --version 5.6.0 syncier-jenkins",
-				"helm template jenkins syncier-jenkins-5.6.0.tgz --namespace jenkins --include-crds --skip-tests --values values1.yaml --values values2.yaml --output-dir .",
+				"helm fetch --repo https://hub.syncier.cloud/chartrepo/library --version 5.6.0 --destination /temp/helmt-123 syncier-jenkins",
+				"helm template jenkins /temp/helmt-123/temp/helmt-123/chart-1.0.0.tgz --namespace jenkins --include-crds --skip-tests --values values1.yaml --values values2.yaml --output-dir .",
```
Comment (torstenwalter) [inline pkg/helmt/helmt_test.go:201]: Same here

File: pkg/helmt/helmt_test.go (hunk 1 above)
Comment (a-hat) [reply pkg/helmt/helmt_test.go:190]: Definitely, thanks.

File: pkg/helmt/helmt.go [hunk 3]
```diff
@@ -130,18 +131,38 @@ func template(name string, chart string, values []string, namespace string, skip
...
+	args = append(args, "--destination", d)
 	if username != "" {
 		args = append(args, "--username", username)
 	}
 	if password != "" {
 		args = append(args, "--password", password)
 	}
 	args = append(args, chart)
-	return execute("helm", execOpts{}, args...)
+	err = execute("helm", execOpts{}, args...)
```
Comment (torstenwalter) [inline pkg/helmt/helmt.go:151]: From here to line 163 one could extract a method which just returns the filename of the downloaded chart

File: pkg/helmt/helmt.go [hunk 4]
```diff
@@ -130,18 +131,38 @@ func template(name string, chart string, values []string, namespace string, skip
 	return execute("helm", execOpts{}, args...)
 }
 
-func fetch(repository, chart, version string, username, password string) error {
+func fetch(repository, chart, version string, username, password string) (string, error) {
+	d, err := tempDir()
```
Comment (torstenwalter) [inline pkg/helmt/helmt.go:135]: You could add a defer statement to delete the temp dir.
But I am not sure about this as one could also call it a feature to be able to copy and paste the helm template command and use it with the downloaded file.

**Review by torstenwalter — APPROVED** [review]
Looks good. Thanks for implementing it that quickly.

**Pull request merged**